### PR TITLE
fix: catch uncaught exception with deserializing config

### PR DESCRIPTION
**Pull request by jsalaber**
# Summary

For cases when exception is not a `DVCException`, it was throwing an argument exception when deserializing the config string into a config object. Added an extra catch to catch any system exceptions not of type `DVCException`

### CODE REVIEW

File: DevCycle.SDK.Server.Local/ConfigManager/EnvironmentConfigManager.cs [hunk 1]
```diff
@@ -205,6 +205,9 @@ private async Task FetchConfigAsyncWithTask()
 
                 logger.LogError(finalError.ErrorResponse.Message);
                 dvcEventArgs.Errors.Add(finalError);
+            } catch (Exception e) {
```
Comment (ajwootto) [inline DevCycle.SDK.Server.Local/ConfigManager/EnvironmentConfigManager.cs]: is there no more specific exception you can be catching here? this basically catches any error at all

File: DevCycle.SDK.Server.Local/ConfigManager/EnvironmentConfigManager.cs [hunk 2]
```diff
@@ -205,6 +205,9 @@ private async Task FetchConfigAsyncWithTask()
...
+                logger.LogError(e.Message);
+                dvcEventArgs.Errors.Add(new DVCException(HttpStatusCode.InternalServerError, new ErrorResponse(e.Message)));
```
Comment (ajwootto) [inline DevCycle.SDK.Server.Local/ConfigManager/EnvironmentConfigManager.cs]: is there a version of DVCException that doesn't have a "status code"? doesn't really make sense to be setting an http status code here

File: DevCycle.SDK.Server.Local/ConfigManager/EnvironmentConfigManager.cs (hunk 1 above)
Comment (jsalaber) [reply DevCycle.SDK.Server.Local/ConfigManager/EnvironmentConfigManager.cs]: yep i can use ArgumentNullException

File: DevCycle.SDK.Server.Local/ConfigManager/EnvironmentConfigManager.cs [hunk 3]
```diff
@@ -205,6 +205,9 @@ private async Task FetchConfigAsyncWithTask()
 
                 logger.LogError(finalError.ErrorResponse.Message);
                 dvcEventArgs.Errors.Add(finalError);
+            } catch (System.ArgumentNullException e) {
```
Comment (ajwootto) [inline DevCycle.SDK.Server.Local/ConfigManager/EnvironmentConfigManager.cs:208]: perhaps move the call to SetConfig to its own try...catch thats looking for this specific error, and comment what it means when its caught?

Comment (jsalaber) [reply DevCycle.SDK.Server.Local/ConfigManager/EnvironmentConfigManager.cs:208]: I think having its own try catch makes it harder to read and bulkier, I could probably just slap some comments in this catch explaining what could throw a `ArgumentNullException`

**Review by ajwootto — APPROVED** [review]

**Pull request merged**